### DISCUSSION TITLE
Issue with Windows 10 Firewall (Python code works, pyinstaller exe is blocked)

### DISCUSSION THREAD

**Discussion by carlos-vl** [pyinstaller/pyinstaller#5900] [PyInstaller]
I am using [ray.io](https://github.com/ray-project/ray) in an application, and I would like to generate an .exe for windows users using pyinstaller. I had to write a hook, but after that the folder with the .exe and corresponding libraries is properly generated. The problem comes when running the generated .exe, the network services started by [ray.io](https://github.com/ray-project/ray) are blocked by the Windows 10 firewall (which I cannot change due to the company policy).  

I know this is an issue of the firewall and not of pyinstaller, but my question is why do those same services are not blocked when I run everything with the python interpreter, but when using the pyinstaller exe everything is blocked. If they are not blocked by the firewall when using the python interpreter, I think there should be a way of getting them to run using the exe generated with pyinstaller.  

Anyone can give some input on this? Thanks in advance!

**Comment by rokm** [comment]
> why do those same services are not blocked when I run everything with the python interpreter

When you run code from python interpreter, you're operating under rules that apply to the `python.exe` program executable. Perhaps there are explicit rules in your firewall policy, or it could be related to the fact that `python.exe` is signed (and may be trusted by the firewall).

**Reply by rokm** [reply]
I suppose to test the latter, you could either try compiling python from source (so you end up with unsigned binary), or remove the signature from the existing binary (after backing it up) and see if things still work.
